stdlib/ballerina-http/src/main/ballerina/net.http.authadaptor/authn-filter.bal

```diff
@@ -0,0 +1,147 @@
+@Param {value:"authorized: authorization status for the request"}
+@Return {value:"FilterResult: Authorization result to indicate if the request can proceed or not"}
+function createAuthnResult (boolean authenticated) returns (http:FilterResult) {
+    http:FilterResult requestFilterResult;
+    if (authenticated) {
+        requestFilterResult = {canProceed:true, statusCode:200, message:"Successfully authenticated"};
+    } else {
+        requestFilterResult = {canProceed:false, statusCode:401, message:"Authentication failure"};
+    }
+    return requestFilterResult;
+}
+
+@Description {value:"Checks if the resource is secured"}
+@Param {value:"context: FilterContext object"}
+@Return {value:"boolean: true if the resource is secured, else false"}
+function isResourceSecured (http:FilterContext context) returns (boolean) {
+    // get authn details from the resource level
+    auth:Authentication authAnn = getAuthnAnnotation(
+                                  internal:getResourceAnnotations(context.serviceType, context.resourceName));
+    match authAnn {
+        auth:Authentication authAnnotation => {
+            return authAnnotation.enabled;
+        }
+        error err => {
+            // if not found at resource level, check in the service level
+            authAnn = getAuthnAnnotation(internal:getServiceAnnotations(context.serviceType));
+            match authAnn {
+                auth:Authentication authAnnotation => {
+                    return authAnnotation.enabled;
+                }
+                error err => {
+                    // if still authentication annotation is null, means the user has not specified that the service
+                    // should be secured. However since the authn filter has been engaged, need to authenticate.
+                    return true;
+                }
+            }
+        }
+    }
+    return false;
+}
+
+@Description {value:"Tries to retrieve the annotation value for authentication hierarchically - first from the resource
+level
+and then from the service level, if its not there in the resource level"}
+@Param {value:"annData: array of annotationData instances"}
+@Return {value:"Authentication: Authentication instance if its defined, else null"}
+function getAuthnAnnotation (internal:annotationData[] annData) returns (auth:Authentication|error) {
+    if (lengthof annData == 0) {
+        error err = {message:"empty annotationData"};
+        return err;
+    }
+    internal:annotationData authAnn = {};
+    foreach ann in annData {
+        if (ann.name == AUTH_ANN_NAME && ann.pkgName == AUTH_ANN_PACKAGE) {
+            authAnn = ann;
+            break;
+        }
+    }
+    if (lengthof authAnn == 0) {
+        // no annotation found for ballerina.auth:config
+        error err = {message:"no annotation for ballerina.auth:Config"};
+        return err;
+    }
+    var authConfig = <auth:AuthConfig> authAnn.value;
+    match authConfig {
```

Comment:
You don't need match here. You can use ?=

Reply:
fixed.